modules/openthread/platform/radio.c

```diff
@@ -920,11 +920,13 @@ void otPlatRadioSetRxOnWhenIdle(otInstance *aInstance, bool aRxOnWhenIdle)
 		.rx_on_when_idle = aRxOnWhenIdle
 	};
 
-	ARG_UNUSED(aInstance);
-
 	LOG_DBG("RxOnWhenIdle=%d", aRxOnWhenIdle ? 1 : 0);
 
 	radio_api->configure(radio_dev, IEEE802154_CONFIG_RX_ON_WHEN_IDLE, &config);
+
+	if (!aRxOnWhenIdle) {
+		ARG_UNUSED(otPlatRadioSleep(aInstance));
+	}
```

Comment:
Thanks @edmont.

The "RxOnWhenIdle" flag can be changed by various triggers within the OT stack. For instance, the `otThreadSetLinkMode()` API, called by higher-level code controlling the OT stack, can switch the device mode, making it sleepy or Rx-on at any point in time.

Therefore, we should consider that `otPlatRadioSetRxOnWhenIdle()` could also be invoked at any time, while the radio may be in different states. For example, the radio could be in the middle of frame transmission or performing an energy scan.

Would calling `otPlatRadioSleep()` in all cases be appropriate? For instance, if the radio is mid-transmission, the OT stack expects a "send done" callback after a successful `otPlatRadioTransmit()`. Will this behave correctly after calling `otPlatRadioSleep()` in such a case?

Reply:
Discussion in upstream PR.